torch/distributed/device_mesh.py

```diff
@@ -823,11 +846,17 @@ def from_group(
                     (_get_group_tag(group), group_ranks, group.group_name)
                 ]
                 return device_mesh
+
+            # nD scenario
             groups = list(group)
             if len(groups) == 0:
                 raise ValueError("Expects at least one ProcessGroup to be passed")
             if mesh is None:
```

Comment:
Not a blocker but I have a question around this condition, do we really need this one? I mean can we infer this? we can just assume the index of passed in PG as the mesh no? Also, if mesh is passed in, we don't need mesh_dim_names right? They are OR not AND?

Reply:
Disccued offline regarding keeping `mesh` tensor required for now but this is subject to change.